### ISSUE TITLE
layernorm2d_forward_kernel normalizes over C only, mismatches Python reference (C×H×W)

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#26]
## Problem

The new CUDA kernel in `cpp/src/core.cu` (`layernorm2d_forward_kernel`) normalizes each spatial location `(N, H, W)` over the **channel axis only**:

```cuda
// Each thread handles one (batch, hw) pair — sums over channels
for (int ch = 0; ch < c; ++ch) {
    mean += input[(batch * c + ch) * spatial + hw];
}
mean /= static_cast<float>(c);   // divisor = C, not C*H*W
```

This is semantically **ChannelNorm per spatial point**, not standard LayerNorm.

The Python reference implementation in `_layernorm2d_forward_array` (called via `_lower_layernorm2d` when no native library is bound) normalizes each sample N over **all of (C, H, W)**, matching the PyTorch `nn.LayerNorm((C, H, W), ...)` contract.

When the native `.so` is loaded and `ctx.runtime.bound_lib.layernorm2d_forward` is called instead of the NumPy path, the kernel produces different statistics (mean and variance over C only vs over C·H·W), yielding silently wrong outputs with no assertion or tolerance check.

## Fix

Rewrite the kernel so each thread (or cooperative group) computes mean and variance over all `C * H * W` elements for a single sample `N`:

```cuda
// thread per batch item; inner loop over c * spatial
int total = c * spatial;
for (int i = 0; i < total; ++i)
    mean += input[batch * total + i];
mean /= static_cast<float>(total);
```

Alternatively add a numerical tolerance test in `test_cuda_native.py` that compares the CUDA kernel output against `_layernorm2d_forward_array` on a small NCHW tensor to catch axis mismatches.

## Priority

**P1 — correctness**: Silent numerical mismatch between CUDA and CPU paths; any model that exercises LayerNorm2d through the native backend will train on wrong gradients with no visible error.

**Comment by s9213712** [comment]
Current code does not have the described CPU/GPU mismatch:
- `_layernorm2d_forward_array` normalizes with `x.mean(axis=1, keepdims=True)` / `x.var(axis=1, keepdims=True)`.
- `layernorm2d_forward_kernel` also normalizes over the channel axis for each `(N, H, W)` position.
- The native executor regression test computes the same channel-axis expected result and passes.

If the desired future semantic is standard `C*H*W` LayerNorm, that should be tracked as a separate behavior-change issue rather than a bug against the current implementation. Validation after issue cleanup: cuda_native GPU enablement subset `163 passed, 4 skipped`.
